### ISSUE TITLE
UnusedFormalParameter should not complain about early parameters if later ones are used

### ISSUE THREAD

**Issue by robballou** [phpmd/phpmd#202]
Often, functions that receive a set of parameters as a callback or similar. If the code then uses one of the later parameters (but not the first one), an UnusedFormParameter is thrown but there's nothing that can be done about it.

Example:

``` php
function my_callback($thing, $thing2) {
    return is_string($thing2);
}
```

`$thing` is needed to get the parameter `$thing2`. Is there a way to fix this? Thanks.

**Comment by ravage84** [comment]
The way explained in https://github.com/phpmd/phpmd/issues/180#issuecomment-305940455 could be a viable solution in such cases. Moving to Future, though.

**Comment by AJenbo** [comment]
Please use this solution: https://github.com/phpmd/phpmd/pull/1193
